tests/test_harvesters.py

```diff
@@ -138,3 +170,49 @@ def test_copernicus_sentinel_harvester(self):
                                                            username='test', password='TEST')
         self.assertIsInstance(harvester._current_crawler, crawlers.CopernicusOpenSearchAPICrawler)
         self.assertIsInstance(harvester._ingester, ingesters.CopernicusODataIngester)
+
+    def test_OSISAF_harvester_exrtra_excludes(self):
+        """ extra excludes should have passed by the excludes as a list in configuration file.
+        Otherwise, accossiated error must be raised """
+        harvester = harvesters.OSISAFHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1,
+                                               excludes=['ease', '_sh_polstere', ])
+        self.assertTrue(harvester.crawler.EXCLUDE, ['/thredds/', 'http', 'ease', '_sh_polstere'])
+        with self.assertRaises(HarvesterConfigurationError):
+            harvester = harvesters.OSISAFHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1,
+                                                   excludes='ease')
+
+
+class HarvesterExceptTestCase(unittest.TestCase):
+    def tearDown(self):
+        mock.patch.stopall()
+
+    def test_except_create_crawler(self):
+        """shall return exception in the case of incorrect class of crawler"""
+        class test_class_harvester(harvesters.WebDirectoryHarvester):
+            ingester = ingesters.DDXIngester
+        with self.assertRaises(HarvesterConfigurationError):
+            test_class_harvester()
+
+    def test_except_create_crawler2(self):
+        """ shall return exception in the case of incorrect class of ingester """
+        class test_class_harvester4(harvesters.WebDirectoryHarvester):
+            crawler = crawlers.OpenDAPCrawler
+        with self.assertRaises(HarvesterConfigurationError):
+            test_class_harvester4()
+
+    @mock.patch("geospaas_harvesting.harvesters.WebDirectoryHarvester._create_crawlers")
+    def test_except_create_ingester(self,mock_create_crawlers):
+        """ shall return exception in the case of incorrect class of ingester """
+        class test_class_harvester2(harvesters.WebDirectoryHarvester):
+            crawler = crawlers.OpenDAPCrawler('', time_range=(None, datetime(2019, 2, 20)))
+        with self.assertRaises(HarvesterConfigurationError):
+            test_class_harvester2()
+        mock_create_crawlers.stop()
+        self.addCleanup(mock_create_crawlers.stop)
```

Comment:
You don't need to stop the patching when you use the decorator.
(and either of these lines would be enough, you don't need both)

Reply:
Done in the next commit